contracts/src/libraries/Transcript.sol

```diff
@@ -40,29 +41,39 @@ library Transcript {
         self.transcript = abi.encodePacked(self.transcript, comm.x, comm.y);
     }
 
-    function getAndAppendChallenge(TranscriptData memory self)
-        internal
-        pure
-        returns (uint256 ret)
-    {
-        bytes memory transcript = self.transcript;
+    // 1. state = hash(state | transcript)
+    // 2. transcript = Vec::new()
+    // 3. challenge = bytes_to_field(state)
+    //
+    // Note: every challenge generation is implicitly domain-separated, thus it's safe
+    // to call it multiple times in a row (e.g. multiple challenges in a single round)
+    // to get multiple different challenges.
+    function getChallenge(TranscriptData memory self) internal pure returns (uint256 ret) {
+        // memory layout:
+        // offset 0x00: state
+        // offset 0x20: pointer to transcript (i.e. 0x40)
+        // offset 0x40: length of transcript
```

Comment:
Not sure this is the case even though I like a lot the documentation. When I change the line

`let lenPtr := mload(add(self, 0x20))`

for

`let lenPtr := add(self, 0x40)`

The tests don't pass anymore. Am I missing something?

Reply:
done in [c28459c](https://github.com/EspressoSystems/espresso-sequencer/pull/1829/commits/c28459cbc1655b33e5ead1dc14d58ba686ec4d31)